From 8ed8e05e2b7648b06f9d8692b3d5c27d23581d22 Mon Sep 17 00:00:00 2001
From: Mihails Strasuns <mihails.strasuns.contractor@sociomantic.com>
Date: Fri, 17 Jun 2016 20:48:41 +0300
Subject: [PATCH 07/17] Disable stomping prevention check by default

Based on practical investigation of first ported apps it is not
important enough to force on everyone.
---
 src/rt/lifetime.d | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/rt/lifetime.d b/src/rt/lifetime.d
index 56382e06..afacd651 100644
--- a/src/rt/lifetime.d
+++ b/src/rt/lifetime.d
@@ -241,9 +241,8 @@ void stomping_prevention_trigger_nonpure ( ) nothrow
     import core.stdc.stdlib : getenv;
     import core.stdc.string : strcmp;
     char* flag = getenv("ALLOW_STOMPING_PREVENTION".ptr);
-    if (flag && (strcmp(flag, "1") == 0))
-        return;
-    assert(false, "Stomping prevention has been triggerred");
+    if (flag && (strcmp(flag, "0") == 0))
+        assert(false, "Stomping prevention has been triggerred");
 }
 
 /**
-- 
2.11.0

